vaadin-spreadsheet/src/main/java/com/vaadin/addon/spreadsheet/DefaultHyperlinkCellClickHandler.java

```diff
@@ -39,48 +46,117 @@
 public class DefaultHyperlinkCellClickHandler implements
         HyperlinkCellClickHandler {
 
-    private static final DefaultHyperlinkCellClickHandler instance = new DefaultHyperlinkCellClickHandler();
+    private static final Logger LOGGER = Logger
+            .getLogger(DefaultHyperlinkCellClickHandler.class.getName());
 
-    protected DefaultHyperlinkCellClickHandler() {
+    private final Spreadsheet spreadsheet;
+    private Navigator navigator;
+    private HyperlinkOpenStyle openStyle;
+
+    /**
+     * @param spreadsheet
+     */
+    public DefaultHyperlinkCellClickHandler(Spreadsheet spreadsheet) {
+        this(spreadsheet, null, HyperlinkOpenStyle.NewTab);
+    }
+
+    /**
+     * @param spreadsheet
+     * @param navigator
+     *            to use, if fragment links are present
+     * @param openStyle
+     *            defaults to {@link HyperlinkOpenStyle#NewTab} if null
+     */
+    public DefaultHyperlinkCellClickHandler(Spreadsheet spreadsheet,
+            Navigator navigator, HyperlinkOpenStyle openStyle) {
+        this.spreadsheet = spreadsheet;
+        this.navigator = navigator;
```

Comment:
Is it necessary to keep the reference to the navigator? Can't we get it using `UI.getCurrent().getNavigator()`?

Reply:
I had forgotten this because the app I referenced for the code wasn't using it (but it should).  I've removed the explicit navigator in favor of the "current" one from the UI, if present.

Also fixed the failing unit test, which was indirectly referencing a private method whose signature changed.